specifyweb/frontend/js_src/lib/components/DataModel/businessRuleDefs.ts

```diff
@@ -156,6 +157,16 @@ export const businessRuleDefs: MappedBusinessRuleDefs = {
           new tables.CollectingEvent.Resource()
         );
       }
+
+      // Set the default CoType
+      if (typeof schema.defaultCollectionObjectType === 'number')
```

Comment:
You still need the `collectionObject.get('collectionObjectType') === undefined` check here (and you can include `null` as well if that is part of the requirements): otherwise we overwrite the collectionObjectType on the collectionObject if the type is different from the default collectionObjectType.

Reply:
Is it not safe to assume a CO will always be initialized with an undefined COType?

Reply:
Yes, for new COs it's safe to assume that the COType will be undefined if it is not set. 

`customInit` as a rule is called regardless of whether the SpecifyResource is new or not: specifically, it's called whenever businessrules are attached to a resource, which happens when BackBone initializes a resource. 

https://github.com/specify/specify7/blob/92c94292c6d70fb867994d88ca503ea56572ea8f/specifyweb/frontend/js_src/lib/components/DataModel/businessRules.ts#L54-L63

https://github.com/specify/specify7/blob/92c94292c6d70fb867994d88ca503ea56572ea8f/specifyweb/frontend/js_src/lib/components/DataModel/businessRules.ts#L462-L468

https://github.com/specify/specify7/blob/92c94292c6d70fb867994d88ca503ea56572ea8f/specifyweb/frontend/js_src/lib/components/DataModel/resourceApi.ts#L128

So `customInit` is also called on CollectionObjects which are not new, and they can have a non-undefined collectionObjectType